File: src/main/java/chat/controller/ChatController.java
package chat.controller;

import chat.controller.dto.Message;
import chat.controller.dto.OutputMessage;
import org.springframework.messaging.handler.annotation.DestinationVariable;
import org.springframework.messaging.handler.annotation.MessageMapping;
import org.springframework.messaging.handler.annotation.SendTo;
import org.springframework.stereotype.Controller;

@Controller
public class ChatController {

    @MessageMapping("/chat/{topic}")
    @SendTo("/topic/messages")
    public OutputMessage send(@DestinationVariable("topic") String topic,
                              Message message) throws Exception
    {
        return new OutputMessage(message.getFrom(), message.getText(), topic);
    }

}

File: src/main/kotlin/chatapp/controller/ChatController.kt
package chatapp.controller

import chat.dto.Message
import chat.dto.OutputMessage
import org.springframework.messaging.handler.annotation.DestinationVariable
import org.springframework.messaging.handler.annotation.MessageMapping
import org.springframework.messaging.handler.annotation.SendTo
import org.springframework.stereotype.Controller
import java.util.*

@Controller
class ChatController {

    @MessageMapping("/chat/{topic}")
    @SendTo("/topic/messages")
    @Throws(Exception::class)
    fun send(@DestinationVariable("topic") topic: String, message: Message): OutputMessage {
        return OutputMessage(message.from, message.text, topic)
    }

    //data class Message(val from: String, val text: String) todo json mapping not working

    //data class OutputMessage(val from: String, val text: String, val topic: String, val time: Date)
}